src/picklescan/scanner.py

```diff
@@ -307,12 +327,35 @@ def scan_pickle_bytes(data: IO[bytes], file_id, multiple_pickles=True) -> ScanRe
     return _build_scan_result_from_raw_globals(raw_globals, file_id)
 
 
+# XXX: it appears there is not way to get the byte stream for a given file within the 7z archive and thus forcing us to unzip to disk before scanning
+def scan_7z_bytes(data: IO[bytes], file_id) -> ScanResult:
+    result = ScanResult([])
+
+    archive = py7zr.SevenZipFile(data, mode="r")
+    file_names = archive.getnames()
+    _log.debug("Files in 7z archive %s: %s", file_id, file_names)
+    rand_id = "".join(random.choice(string.ascii_letters) for _ in range(8))
+    tmp_path = f"/tmp/archive-{rand_id}"
```

Comment:
I would expect writing to the /tmp folder to fail on Windows. It would be better to rely on Python's platform-agnostic temp file management using `NamedTemporaryFile`, `TemporaryDirectory`, and the like. This should also simplify a bit the code (no need to import `random`/`string`, clean up files, etc.).